vendor/engines/split_accounts/app/services/split_accounts/split_order_detail_simulator.rb

```diff
@@ -0,0 +1,90 @@
+module SplitAccounts
+
+  # TODO: maybe refactor some of this into SplitSimulator
+  # TODO: we can refactor this code to fewer lines and iterate fewer times
+  class SplitOrderDetailSimulator
```

Comment:
I'm not crazy about `Simulator` either.  I was trying to convey that the resulting order details were `read-only`.